lib/features/upload/domain/model/upload_attachment.dart

```diff
@@ -57,12 +60,16 @@ class UploadAttachment with EquatableMixin {
       }
 
       _updateEvent(Right(SuccessAttachmentUploadState(uploadTaskId, attachment, fileInfo)));
-    } catch (e) {
-      logError('UploadAttachment::upload():ERROR: $e');
-      if (e is DioError && e.type == DioErrorType.cancel) {
+    } catch (error, stackTrace) {
+      logError('UploadAttachment::upload():ERROR: $error');
+      if (error is DioError && error.type == DioErrorType.cancel) {
         _updateEvent(Left(CancelAttachmentUploadState(uploadTaskId)));
       } else {
-        _updateEvent(Left(ErrorAttachmentUploadState(uploadId: uploadTaskId)));
+        try {
+          exceptionThrower.throwException(error, stackTrace);
```

Comment:
I found that it was caught in `lib/features/upload/data/datasource_impl/attachment_upload_datasource_impl.dart`. So we only need to throw exception here

Reply:
No, we execute multiple upload functions at the same time, so the try/catch in the datasource only catches the exception at the time the datasource finishes calling. Moreover, the `upload` function inside has been try/catch to update the result for StreamController, so we need to handle the exception in this function.